spec.html

```diff
@@ -8450,6 +8450,17 @@ <h1>Module Namespace Exotic Objects</h1>
               A List containing the String values of the exported names exposed as own properties of this object. The list is ordered as if an Array of those String values had been sorted using `Array.prototype.sort` using *undefined* as _comparefn_.
             </td>
           </tr>
+          <tr>
+            <td>
+              [[HostDefined]]
+            </td>
+            <td>
+              Any, default value is *undefined*.
+            </td>
+            <td>
+              Field reserved for use by host environments that need to associate additional information with a Module Namespace Exotic Object.
```

Comment:
I'm a bit confused why this is needed - implementations can always add any number of internal slots (or side-channel metadata) to any value.

Reply:
I've applied this in https://github.com/nodejs/dynamic-modules/blob/master/spec.html#L203.

The nice thing about this is that it avoids having to declare the field in the dynamic modules spec, retaining the spec separation of simply implementing a subclass of module record and not changing any other objects or fields.

Reply:
Other suggestions welcome too.

Reply:
It seems like if the only spec to use it is the dynamic modules spec, then that's the place it should be declared :-/

Reply:
Are you saying that a spec can add fields to any object in the ECMA-262 spec while remaining valid against ECMA-262? I thought the point of [[HostDefined]] was to clarify these cases?

Reply:
Although technically hosts can do crazy things, it's much more friendly to spec readers and to implementers if integration points are called out explicitly and made to interoperate in the same way they would in implementations. I strongly support adding [[HostDefined]] fields where appropriate in this way to increase the fidelity and readability of the spec's layering.